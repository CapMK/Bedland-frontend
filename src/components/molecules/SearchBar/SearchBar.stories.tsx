import { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import SearchBar from './SearchBar';

const meta = {
  title: 'Molecules/SearchBar',
  component: SearchBar,
  tags: ['autodocs'],
  args: { placeholder: 'Search flats...' },
  argTypes: {
    placeholder: {
      description: 'Placeholder value',
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof SearchBar>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
