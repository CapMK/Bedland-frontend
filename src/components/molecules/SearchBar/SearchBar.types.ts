export const filterOptions = ['name'] as const;

export type ExeListItemType = { id: number; name: string };

export type SearchBarProps = { placeholder: string };
