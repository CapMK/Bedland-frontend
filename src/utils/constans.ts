export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
