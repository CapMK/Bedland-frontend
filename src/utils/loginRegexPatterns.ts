import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
