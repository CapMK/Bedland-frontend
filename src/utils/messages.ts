const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
