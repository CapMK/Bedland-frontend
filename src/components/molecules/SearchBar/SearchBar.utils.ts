import { ExeListItemType, filterOptions } from './SearchBar.types';

export const STATIC_VALUES = {
  iconSize: 20,
  containerPaddingLeft: 14,
  contentGap: 9,
};

export const exeList: ExeListItemType[] = [
  { id: 1, name: 'Melchior' },
  { id: 2, name: 'Bartek' },
  { id: 3, name: 'Pogruqar' },
  { id: 4, name: 'Marcinek' },
  { id: 5, name: 'SzopPracz' },
];

export const searchFilter = (
  searchValue: string,
  list: ExeListItemType[],
  searchBy: (typeof filterOptions)[number] = 'name',
) => {
  const searchValueLowerCase = searchValue.toLowerCase();
  const filteredList = searchValue
    ? list.filter((item) => item[searchBy].toLowerCase().includes(searchValueLowerCase))
    : list;
  return filteredList;
};
