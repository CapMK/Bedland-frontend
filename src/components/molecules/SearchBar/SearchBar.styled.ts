import styled from 'styled-components';
import initialTheme from 'theme/initialTheme';
import { STATIC_VALUES } from './SearchBar.utils';

const searchBarTypography = {
  ...initialTheme.typography.dataInputsAndTooltips,
};

export const SearchBarContainer = styled.div<{
  ifInputFocused: boolean;
  ifDropdownVisible: boolean;
}>`
  align-items: center;
  background-color: ${initialTheme.palette.white};
  border: 1px solid
    ${({ ifInputFocused }) =>
      ifInputFocused ? initialTheme.palette.blueDark : initialTheme.palette.white};
  border-bottom: 1px solid
    ${({ ifDropdownVisible, ifInputFocused }) =>
      // eslint-disable-next-line no-nested-ternary
      ifDropdownVisible
        ? initialTheme.palette.sidebarTabs
        : ifInputFocused
        ? initialTheme.palette.blueDark
        : 'transparent'};
  border-radius: ${({ ifDropdownVisible }) => (ifDropdownVisible ? '8px 8px 0 0' : '8px')};
  display: flex;
  justify-content: space-between;
  padding: 0.375rem
    ${({ ifDropdownVisible, ifInputFocused }) =>
      ifDropdownVisible || ifInputFocused ? 0.6875 : 0}rem
    0.375rem ${STATIC_VALUES.containerPaddingLeft / 16}rem;
  width: 17.5rem;
`;

export const SearchBarContent = styled.div`
  align-items: center;
  background-color: ${initialTheme.palette.white};
  display: flex;
  flex-wrap: nowrap;
  gap: ${STATIC_VALUES.contentGap / 16}rem;
  height: 1.75rem;
  justify-content: start;
`;

export const Input = styled.input`
  ${searchBarTypography}
  width: 100%;
`;

export const DropdownContainer = styled.div`
  position: relative;
`;

export const DropdownList = styled.ul<{ ifDropdownVisible: boolean }>`
  background-color: ${({ ifDropdownVisible }) =>
    ifDropdownVisible ? initialTheme.palette.white : 'transparent'};
  border: 1px solid
    ${({ ifDropdownVisible }) =>
      ifDropdownVisible ? initialTheme.palette.blueDark : 'transparent'};
  border-radius: 0 0 8px 8px;
  border-top: 1px solid
    ${({ ifDropdownVisible }) =>
      ifDropdownVisible ? initialTheme.palette.sidebarTabs : 'transparent'};
  position: absolute;
  width: 100%;
`;

export const DropdownTextContainer = styled.li`
  cursor: pointer;
  display: flex;
  padding: 0.5rem 0 0.6875rem
    ${(STATIC_VALUES.iconSize + STATIC_VALUES.containerPaddingLeft + STATIC_VALUES.contentGap) /
    16}rem;
`;

export const BoldedLetter = styled.span`
  font-weight: bold;
`;
