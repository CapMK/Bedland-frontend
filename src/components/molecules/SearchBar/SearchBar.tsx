import { ChangeEvent, FC, useState, useEffect, useRef } from 'react';
import Icon from 'components/atoms/Icon/Icon';
import Typography from 'components/atoms/Typography/Typography';
import Button from 'components/atoms/Button/Button';
import * as Styled from './SearchBar.styled';
import { STATIC_VALUES, exeList, searchFilter } from './SearchBar.utils';
import { ExeListItemType, SearchBarProps } from './SearchBar.types';

const SearchBar: FC<SearchBarProps> = ({ placeholder }) => {
  const [isFocused, setIsFocused] = useState<boolean>(false);
  const [searchValue, setSearchValue] = useState('');
  const [ifDropdownVisible, setIfDropdownVisible] = useState(false);
  const dropdownRef = useRef<HTMLDivElement>(null);

  const matches = searchFilter(searchValue, exeList);

  const handleFocus = () => {
    setIsFocused(true);
    // eslint-disable-next-line no-unused-expressions
    searchValue.split('').length > 0 && setIfDropdownVisible(true);
  };
  const handleOnClickOutside = () => {
    setIsFocused(false);
    setIfDropdownVisible(false);
  };
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setSearchValue(e.target.value);
    if (!ifDropdownVisible) {
      setIfDropdownVisible(true);
    }
    if (e.target.value === '') setIfDropdownVisible(false);
  };
  const handleSelectItem = (item: ExeListItemType) => {
    setSearchValue(item.name);
    setIfDropdownVisible(false);
    setIsFocused(false);
  };
  const handleRemoveButton = async () => {
    setIfDropdownVisible(false);
    setSearchValue('');
    setIsFocused(false);
  };

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent): void => {
      if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
        handleOnClickOutside();
      }
    };
    document.addEventListener('click', handleClickOutside, true);
    return () => {
      document.removeEventListener('click', handleClickOutside, true);
    };
  }, []);

  return (
    <div ref={dropdownRef}>
      <Styled.SearchBarContainer ifInputFocused={isFocused} ifDropdownVisible={ifDropdownVisible}>
        <Styled.SearchBarContent>
          <Icon
            name='search'
            isActive={false}
            size={STATIC_VALUES.iconSize}
            color={isFocused ? 'blueDark' : 'inputGrey'}
          />
          <Styled.Input
            data-testid='SearchBarInput'
            onFocus={handleFocus}
            value={searchValue}
            onChange={(e) => handleChange(e)}
            placeholder={isFocused ? '' : placeholder}
          />
        </Styled.SearchBarContent>
        {ifDropdownVisible ? (
          <Button wrapper style={{ marginRight: '0.6875rem' }}>
            <Button
              variant='icon'
              icon='remove'
              iconColor='greyDark'
              onClick={handleRemoveButton}
            />
          </Button>
        ) : null}
      </Styled.SearchBarContainer>
      <Styled.DropdownContainer>
        {ifDropdownVisible && (
          <Styled.DropdownList ifDropdownVisible={ifDropdownVisible} data-testid='dropdownList'>
            {exeList && matches.length > 0 ? (
              matches.map((listItem: ExeListItemType) => (
                // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
                <Styled.DropdownTextContainer
                  key={listItem.id}
                  onClick={() => handleSelectItem(listItem)}
                >
                  {listItem.name.split('').map((letter, index) =>
                    searchValue
                      .toLowerCase()
                      .split('')
                      .find((searchLetter) => searchLetter === letter.toLowerCase()) ? (
                      <Typography variant='dataInputsAndTooltips' key={`${letter + index}`}>
                        <Styled.BoldedLetter>{letter}</Styled.BoldedLetter>
                      </Typography>
                    ) : (
                      <Typography variant='dataInputsAndTooltips' key={`${letter + index}`}>
                        {letter}
                      </Typography>
                    ),
                  )}
                </Styled.DropdownTextContainer>
              ))
            ) : (
              <Styled.DropdownTextContainer>No results</Styled.DropdownTextContainer>
            )}
          </Styled.DropdownList>
        )}
      </Styled.DropdownContainer>
    </div>
  );
};

export default SearchBar;
