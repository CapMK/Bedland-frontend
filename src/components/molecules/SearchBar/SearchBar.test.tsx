import { cleanup, fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import SearchBar from './SearchBar';

describe('SearchBar', () => {
  afterEach(cleanup);
  describe('SearchBar component', () => {
    const placeholderText = 'Search...';

    const searchBarRender = () =>
      renderWithProviders(
        <div>
          <div data-testid='testDiv'>SomeTest</div>
          <SearchBar placeholder={placeholderText} />, {}
        </div>,
      );
    it('should render correctly', () => {
      const searchBar = searchBarRender();
      expect(searchBar).toMatchSnapshot();
    });

    it('should show the dropdown and the remove button when the search input is focused', () => {
      const { getByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      fireEvent.focus(inputField);
      fireEvent.change(inputField, { target: { value: 'a' } });
      const dropdown = getByTestId('dropdownList');
      expect(dropdown).toBeTruthy();
      const removeIcon = getByTestId('remove-icon');
      expect(removeIcon).toBeTruthy();
    });

    it('should clear input, unfocus input and close dropdown after remove button click ', () => {
      const { getByTestId, queryByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      fireEvent.focus(inputField);
      fireEvent.change(inputField, { target: { value: 'a' } });
      const removeIcon = getByTestId('remove-icon');
      fireEvent.click(removeIcon);

      const dropdown = queryByTestId('dropdownList');
      const removeIconAfterClick = queryByTestId('remove-icon');
      expect(dropdown).toBeFalsy();
      expect(removeIconAfterClick).toBeFalsy();
      expect(inputField).not.toHaveFocus();

      expect(inputField).toHaveValue('');
    });

    it('should hide the dropdown on the outside component click', () => {
      const { getByTestId, queryByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      const testDiv = getByTestId('testDiv');
      fireEvent.focus(inputField);
      fireEvent.click(testDiv);

      const dropdown = queryByTestId('dropdownList');
      expect(dropdown).toBeFalsy();
    });

    it('should show dropdown when focused on the search field with text', () => {
      const { getByTestId, queryByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      const testDiv = getByTestId('testDiv');
      fireEvent.change(inputField, { target: { value: 'bar' } });
      fireEvent.click(testDiv);
      fireEvent.focus(inputField);

      const dropdown = queryByTestId('dropdownList');
      expect(dropdown).toBeTruthy();
    });

    it('should show filtered list items in the dropdown when user types in the search field', () => {
      const { getByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      fireEvent.change(inputField, { target: { value: 'a' } });

      const listItemContainers = getByTestId('dropdownList').querySelectorAll('li');
      expect(listItemContainers.length > 0).toBe(true);
    });

    it('should show no results message when user types wrong value in the search field', () => {
      const { getByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      fireEvent.change(inputField, { target: { value: 'xas' } });
      expect(screen.getByText('No results'));
    });

    it('should select a list item when it is clicked on and update the search value', () => {
      const { getByTestId } = searchBarRender();
      const inputField = getByTestId('SearchBarInput');
      fireEvent.focus(inputField);
      fireEvent.change(inputField, { target: { value: 'a' } });
      const listItemContainers = getByTestId('dropdownList').querySelectorAll('li');
      const firstListItemContainer = listItemContainers[0];
      fireEvent.click(firstListItemContainer);

      expect((inputField as HTMLInputElement).value).toBe('Bartek');
    });
  });
});
