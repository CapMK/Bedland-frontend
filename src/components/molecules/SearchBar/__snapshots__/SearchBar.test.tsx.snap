// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > SearchBar component > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-bottom: 1px solid transparent;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.375rem 0rem 0.375rem 0.875rem;
  width: 17.5rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.5625rem;
  height: 1.75rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c2 {
  font-family: Source Sans Pro,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 100%;
}

.c3 {
  position: relative;
}

<body>
    <div>
      <div>
        <div
          data-testid="testDiv"
        >
          SomeTest
        </div>
        <div>
          <div
            class="c0"
          >
            <div
              class="c1"
            >
              <svg
                data-testid="search-icon"
                fill="none"
                height="20"
                viewBox="0 0 16 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  id="mask-search"
                />
                <path
                  d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
                  stroke="#4B4B4B"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.15972"
                />
              </svg>
              <input
                class="c2"
                data-testid="SearchBarInput"
                placeholder="Search..."
                value=""
              />
            </div>
          </div>
          <div
            class="c3"
          />
        </div>
        , 
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-bottom: 1px solid transparent;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.375rem 0rem 0.375rem 0.875rem;
  width: 17.5rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.5625rem;
  height: 1.75rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c2 {
  font-family: Source Sans Pro,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 100%;
}

.c3 {
  position: relative;
}

<div>
    <div>
      <div
        data-testid="testDiv"
      >
        SomeTest
      </div>
      <div>
        <div
          class="c0"
        >
          <div
            class="c1"
          >
            <svg
              data-testid="search-icon"
              fill="none"
              height="20"
              viewBox="0 0 16 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-search"
              />
              <path
                d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
                stroke="#4B4B4B"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.15972"
              />
            </svg>
            <input
              class="c2"
              data-testid="SearchBarInput"
              placeholder="Search..."
              value=""
            />
          </div>
        </div>
        <div
          class="c3"
        />
      </div>
      , 
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
